pkg/object/s3.go

```diff
@@ -374,8 +374,9 @@ func newS3(endpoint, accessKey, secretKey string) (ObjectStorage, error) {
 				region = parseRegion(endpoint)
 			} else {
 				// compatible s3
-				bucketName = hostParts[0]
-				ep = hostParts[1]
+				l := len(hostParts)
+				bucketName = strings.Join(hostParts[:l-4], ".")
```

Comment:
This assume that the endpoint has 4 parts, which could be IP (ipv4 or ipv6)  or DNS, it will not always have 4 parts.